api/routes/news.js

```diff
@@ -2,87 +2,217 @@ const express = require('express');
-    }
-    res.status(200).json({
-        message: 'Single News!',
-        newsId: req.params.id,
-        obj: newsObj
+    const id = req.params.id;
+    News.findById(id)
+    .exec()
+    .then(doc => {
+        if(!doc) {
+            throw new CustomError('Single News Server Error', 500) 
+        } else {
+            res.status(200).json({
+                message: 'Single News!',
+                Details: doc
+            })
+        }        
     })
+    .catch(next)  
 });
 
 app.post('/news', (req, res, next) => {
-    const {error} = validateNews(req.body);    
-    if(error) {
-        throw new CustomError('Not Found', 400)
-    }
-    const newsObj = {
-        id: newsArray.length + 1,
-        name: req.body.name
-    }
-    newsArray.push(newsObj);
-    res.status(201).json({
-        Message: 'Successfully created a News!',
-        newObj: newsObj,
+    const newsObj = new News({
+        name: req.body.name,
+        author: req.body.author,
+        date: req.body.date
+    })
+    newsObj.save()
+    .then(result => {
+        if(!result) {
+            throw new CustomError('Internal Server Error while saving', 500)
+        } else {
+            res.status(201).json({
+                Message: 'Successfully created a News!',
+                createdNews: result,
+            })
+        }        
     })
+    .catch(next)
 });
 
 app.put('/news/:id', (req,res,next) => {
-    const newsObj = newsArray.find(item => item.id === parseInt(req.params.id))
-    if(!newsObj) { 
-        throw new CustomError('The news id was not found', 404)
-    }
-    const {error} = validateNews(req.body);
-    if(error) {
-        throw new CustomError('Bad Request error', 400)
-    }
-    newsObj.name = req.body.name;
-    res.status(200).json({
-        message: 'Successfully updated a News!',
-        updatedObj: newsObj
+    const id = req.params.id;
+    const operations = {};
+    for (const opr of req.body) {
+        operations[opr.propName] = opr.value;
+    } 
+    News.update({_id: id}, { $set: operations})   
+    .exec()
+    .then(result => {
+        if (!result) {
+            throw new CustomError('Internal Server Error - PUT Method', 500)
+        } else {
+            res.status(200).json({
+                message: 'Successfully updated a News!',
+                updatedNews: result
+            })
+        }        
     })
+    .catch(next)
 })
 
 app.delete('/news/:id', (req, res, next) => {
-    const newsObj = newsArray.find(item => item.id === parseInt(req.params.id))
-    if(!newsObj) { 
-        throw new CustomError('The news id was not found', 404)
+    const id = req.params.id;
+    News.remove({_id: id})
+    .exec()
+    .then(result => {
+        if(!result) {
+            throw new CustomError('Unable to Delete News', 500) 
+        } else {
+            res.status(200).json(result)
+        }        
+    })
+    .catch(next)
+});
+
+app.post('/users/register', (req,res) => {
```

Comment:
Lets move users to the separate file of route

Reply:
Separated